src/main/java/com/mercadolivre/desafio_spring/service/PostService.java

```diff
@@ -55,31 +59,39 @@ public UserPromosCountedDTO countPromoByUser(int userId) {
 
     @Override
     public UserPromosDTO getPromosByUser(int userId) {
-        List<Post> listPosts = orderList(this.postRepository.fetchPromosByUser(userId), "");
+        List<Post> listPosts = this.postRepository.fetchPromosByUser(userId);
+        orderList(listPosts, "");
+
         List<PromoPostDTO> promoPostDTOlist = listPosts.stream()
                 .map(Post::toPromoPostDTO)
                 .collect(Collectors.toList());
 
         User user = this.usersRepository.fetchById(userId);
+        validateIfUserExists(user);
 
         return new UserPromosDTO(user.getId(), user.getName(), promoPostDTOlist);
     }
 
-    public List<PostDTO> getOrderedPostList(int userId, String order) {
-        List<Post> orderedPostList = orderList(this.postRepository.fetchPostsByUser(userId), order);
-
-        return orderedPostList.stream()
+    public List<PostDTO> convertPostToDto(List<Post> postList) {
+        return postList.stream()
                 .map(Post::toPostDTO)
                 .collect(Collectors.toList());
     }
 
-    private List<Post> orderList(List<Post> list, String order) {
+    private void orderList(List<Post> list, String order) {
         if(Objects.equals(order, ASC_ORDERING)) {
             list.sort(Comparator.comparing(Post::getDate));
         } else {
             list.sort(Comparator.comparing(Post::getDate).reversed());
         }
+    }
+
+    private List<Post> fetchFollowedPostList(User user) {
+        List<Post> followedPosts = new ArrayList<>();
```

Comment:
o for pode ser uma stream :), como essa:
```
    return user.getFollowed().stream()
                .map(postRepository::fetchPostsByUser)
                .flatMap(List::stream)
                .collect(Collectors.toList())
                ;
```
![Captura de Tela 2021-07-09 às 10 56 40](https://user-images.githubusercontent.com/86322503/125089169-8543bf00-e0a4-11eb-830b-2d5adedc6678.png)

Como vc pode ver na captura, a ide dá uma dica pra gnt do que a stream esta fazendo. Mas basicamente, o mais diferente daí é o flatMap, e ele ajuda a gnt a pegar listas com mais de uma dimensão e juntar ela em apenas uma.

Reply:
<img width="565" alt="Screen Shot 2021-07-09 at 11 05 16" src="https://user-images.githubusercontent.com/69001335/125090373-a35def00-e0a5-11eb-93ba-8c64d6dfda91.png">

corrigido, obrigado pelas dicas =)